Makefile

```diff
@@ -399,6 +407,8 @@ TARGETS = \
 changelog:
 	versionist
 
+webpack: build/js/gui.js
```

Comment:
Feels a little odd having a `webpack` target which depends on a `build/js/gui.js` file, when that file itself is built by running a `webpack` tool. Perhaps the Makefile target could be given a different name to avoid confusion? Perhaps something like `build-webpack`? (and then perhaps the existing `sass` target could be renamed to `build-sass` at the same time? And you could have a `build` target which runs both `build-sass` and `build-webpack`, in a similar style to the `lint` and `test` targets.)

Reply:
Lets handle that in a separate PR (renaming some targets to include a `build` prefix)

Reply:
So, how does the 'sass' and css stuff fit into all of this? It's causing the CI builds to fail, and when I run `make sass` on a local copy of this PR, it complains that "An output directory must be specified when compiling a directory". I can fix that error by editing the Makefile rule to `node-sass lib/gui/app/scss/main.scss > lib/gui/css/main.css` but it then complains about a bunch of incorrect paths.